src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -963,6 +976,39 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger? errorLogger, Cancella
             return existing.WithAdditionalTreeOptions(builder.ToImmutable());
         }
 
+        private CompilerAnalyzerConfigOptionsProvider GetCompilerAnalyzerConfigOptionsProvider(
+            AnalyzerConfigSet? analyzerConfigSet,
+            ImmutableArray<AdditionalText> additionalTextFiles,
+            DiagnosticBag diagnostics,
+            ref Compilation compilation,
+            ImmutableArray<AnalyzerConfigOptionsResult> sourceFileAnalyzerConfigOptions)
+        {
+            var analyzerConfigProvider = CompilerAnalyzerConfigOptionsProvider.Empty;
+            if (Arguments.AnalyzerConfigPaths.Length > 0)
+            {
+                Debug.Assert(analyzerConfigSet is object);
+                analyzerConfigProvider = analyzerConfigProvider.WithGlobalOptions(new DictionaryAnalyzerConfigOptions(analyzerConfigSet.GetOptionsForSourcePath(string.Empty).AnalyzerOptions));
+
+                // TODO(https://github.com/dotnet/roslyn/issues/31916): The compiler currently doesn't support
```

Comment:
```suggestion
                // https://github.com/dotnet/roslyn/issues/31916: The compiler currently doesn't support
```

nit: we try and reserve the word TODO for things which should be fixed or replaced with issue links in a subsequent push to the same PR.